wagtail/management/commands/purge_revisions.py

```diff
@@ -74,11 +77,20 @@ def purge_revisions(days=None, pages=True, non_pages=True):
         purgeable_revisions = purgeable_revisions.filter(created_at__lt=purgeable_until)
 
     deleted_revisions_count = 0
+    protected_error_count = 0
 
     for revision in purgeable_revisions.iterator():
         # don't delete the latest revision
         if not revision.is_latest_revision():
-            revision.delete()
-            deleted_revisions_count += 1
+            try:
+                revision.delete()
+                deleted_revisions_count += 1
+            except ProtectedError:
+                protected_error_count += 1
+
+    sys.stdout.write(
+        f"Deleted {deleted_revisions_count} revisions. "
+        f"Ignored {protected_error_count} revisions due to ProtectedError."
+    )
```

Comment:
**Suggestion:** the code writing to `stdout` should probably be moved up and replace the existing code that is writing to `stdout` here: https://github.com/wagtail/wagtail/blob/1a173af7277676f8bed4548f9515c99b0cfbe808/wagtail/management/commands/purge_revisions.py#L39-L46

As it stands now, this would output duplicate messages which I don't think is desirable.

**Suggestion:** this should use Django's `self.stdout` proxy, not write to `stdout` directly. See the note about this in the Django docmentation: https://docs.djangoproject.com/en/4.2/howto/custom-management-commands/

**Suggestion:** in addition to the above, I believe the current code will write to stdout without a newline (example: `Deleted 10 revisions. Ignored 2 revisions due to ProtectedError`). I feel a newline would make sense, so let's call `write` twice and have it handle the newline for us.

**Suggestion:** finally, may I suggest we tweak the wording a bit? The intent of this suggestion is to make the message more understandable if the person seeing this message is not familiar with ProtectedError and what it signifies.

```suggestion
    self.stdout.write(f"Deleted {deleted_revisions_count} revisions. ")
    self.stdout.write(f"Ignored {protected_error_count} revisions because one or more protected relations exist that prevent deletion.")
```

Reply:
I was confused there because the PR that worked on it before added sys.stdout and the code seemed to return the deleted_revisions_count. I didn't realise the handle function was just above it 😅

Reply:
I've made changes in [ee006fc](https://github.com/wagtail/wagtail/pull/10961/commits/ee006fc18f60a32339af2b5ac10c635e59589045) and [9649d7f](https://github.com/wagtail/wagtail/pull/10961/commits/9649d7f79f63ae1747e017194b163a2a4bd12550). Can you please verify them?

Reply:
Also so far I've been taking suggestions you guys gave and making corresponding changes in my local branch and pushing them to the origin. Should I just do `commit suggestion` instead? Does that have any additional benefits compared to the way I've been doing it?

Reply:
Commit suggestion is fine - just remember to pull the remote to your local branch before making other changes locally. 

Normally, when merging, the core team will add a change log entry / release notes and then squash/rebase anyway. So having a few extra commits in the branch is fine.

Reply:
Do be sure to check the suggestions work of course though.

Reply:
Sure I'll do that 🙌